pkg/agent/flowexporter/utils_test.go

```diff
@@ -96,3 +96,25 @@ func TestPolicyTypeToUint8(t *testing.T) {
 		assert.Equal(t, tc.expectedResult, result)
 	}
 }
+
+func TestLookupProtocolMap(t *testing.T) {
+	for _, tc := range []struct {
+		protocol       string
+		expectedResult uint8
+	}{
+		{"icmp", 1},
+		{"igmp", 2},
+		{"tcp", 6},
+		{"udp", 17},
+		{"ipv6-icmp", 58},
+		{"IPV6-ICMP", 58},
+		{"mockProtocol", 0},
+	} {
+		proto, err := LookupProtocolMap(tc.protocol)
+		if err == nil {
+			assert.Equal(t, tc.expectedResult, proto)
+		} else {
+			assert.Contains(t, err.Error(), "unknown IP protocol specified")
+		}
```

Comment:
it would make more sense as follows:
```go
if tc. expectedResult == 0 {
    assert.ErrorContains(t, err, "unknown IP protocol specified")
} else {
    require.NoError(t, err)
    assert.Equal(t, tc.expectedResult, proto)
}
```

you want the expected result to drive the verification logic, instead of the `err` value returned by the function under test